packages/rest/src/router/routing-table.ts

```diff
@@ -271,6 +295,50 @@ export abstract class BaseRoute implements RouteEntry {
   }
 }
 
+export class StaticAssetsRoute implements RouteEntry {
+  public readonly verb: string = 'get';
+  public readonly path: string = '/*';
+  public readonly spec: OperationObject = {
+    description: 'LoopBack static assets route',
+    'x-visibility': 'undocumented',
+    responses: {},
+  };
+  private _handler: Function;
+  constructor(private _routerForStaticAssets: ExpressRouter) {
+    this._handler = (request: Request, response: Response) => {
+      return new Promise((resolve, reject) => {
+        const onFinished = () => resolve();
+        response.once('finish', onFinished);
+        this._routerForStaticAssets.handle(request, response, (err: Error) => {
+          if (err) {
```

Comment:
When the router called us back, then please remove `onFinished` listener to avoid resolving/rejecting the promise twice and also to avoid a potential (memory) leak.

Reply:
It's a `once`, do we still need to remove it?

Reply:
Yes, because when the static asset cannot be found or a ReadStream cannot be created, then the `finish` event is not fired within the timeframe of this async method.

I guess one could argue that the `finish` method will be eventually fired because the HTTP response will be finished by other parts of the framework eventually, but I find it very brittle and hacky to rely on some other part of our codebase to trigger "finish". See also [Action at a distance](https://en.wikipedia.org/wiki/Action_at_a_distance_%28computer_programming%29) anti-pattern.